src/actions/ui.js

```diff
@@ -61,15 +65,48 @@ export const acUpdateUiProgramStageId = (value, metadata) => ({
     metadata,
 })
 
+const tClearUiProgramDimensions = () => (dispatch, getState) => {
+    const { ui, metadata } = getState()
+    const idsToRemove = ui.layout.columns
+        .concat(ui.layout.filters)
+        .filter((dimensionId) =>
+            DIMENSION_TYPES_PROGRAM.has(metadata[dimensionId].dimensionType)
+        )
+    dispatch(acRemoveUiLayoutDimensions(idsToRemove))
+    dispatch(acRemoveUiItems(idsToRemove))
+}
+
+export const tClearProgramStageDimensions =
+    (stageId) => (dispatch, getState) => {
+        const { ui, metadata } = getState()
+        const program = metadata[ui.program.id]
+        const needsClearing =
+            ui.input.type === OUTPUT_TYPE_EVENT &&
+            program.programType === PROGRAM_TYPE_WITH_REGISTRATION
+
+        if (needsClearing) {
+            const idsToRemove = ui.layout.columns
+                .concat(ui.layout.filters)
+                .filter((id) => {
+                    const idParts = id.split('.')
+                    return idParts.length === 2 && idParts[0] === stageId
+                })
```

Comment:
Maybe this can be simplified using `includes`?
```suggestion
                .filter((id) => id.includes(`${stageId}.`))
```